libraries/delivery_test_kitchen.rb

```diff
@@ -144,6 +147,60 @@ def prepare_kitchen_ec2
       file.run_action(:create)
     end
 
+  #
+  # Specific requirements for azurerm driver
+  #
+  # rubocop:disable AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def prepare_kitchen_azurerm
+    fail 'Kitchen YAML file not found' unless kitchen_yaml?
+
+    # Load secrets from delivery-secrets data bag
+    secrets = get_project_secrets
+    fail 'Could not find secrets for kitchen-azurerm driver' \
+         ' in delivery-secrets data bag.' if secrets['azurerm'].nil?
+
+    # Variables used for configuring and running test kitchen EC2
+    cache                 = delivery_workspace_cache
+    azure_subscription_id = secrets['azurerm']['subscription_id']
+    azure_client_id      = secrets['azurerm']['client_id']
```

Comment:
[nit] the indentation of the `=` are weird.... Either align them or not. (I prefer align as you did 😄)